src/utils/process.rs

```diff
@@ -648,24 +649,37 @@ pub mod tests {
                     TlsState::With(n, args) => TlsState::With(*n + 1, Rc::clone(args)),
                     TlsState::None => TlsState::None,
                     TlsState::Invalid => TlsState::Invalid,
+                    TlsState::Error => TlsState::Error,
                 });
 
                 match old_value {
                     TlsState::Once(args) | TlsState::Scope(args) => Some(args),
                     TlsState::With(n, args) if n < args.len() => Some(args[n].clone()),
                     TlsState::None => None,
-                    TlsState::Invalid | TlsState::With(_, _) => Self::error("get"),
+                    TlsState::Invalid | TlsState::With(_, _) | TlsState::Error => {
+                        Self::error("get");
+                        None
+                    }
                 }
             })
         }
         pub fn are_set() -> bool {
-            FAKE_ARGS.with(|a| *a.borrow() != TlsState::None)
+            FAKE_ARGS.with(|a| *a.borrow() != TlsState::None && *a.borrow() != TlsState::Error)
         }
-        fn error(where_: &str) -> ! {
-            panic!(
-                "test logic error (in {}): wrong FakeParentArgs scope?",
-                where_
-            );
+        fn error(where_: &str) {
+            FAKE_ARGS.with(|a| {
+                let old_value = a.replace(TlsState::Error);
+
+                match old_value {
+                    TlsState::Error => (),
```

Comment:
Just one final "naming things": I think `ErrorAlreadyHandled` or something like that would be a better name, also see `Drop` where another `Error => nothing` line is.

Reply:
I didn't changed any `ErrorAlreadyHandled => nothing` lines because the only case where FAKE_ARGS is set to `ErrorAlreadyHandled` is in the `error` function just before panicking. 
When we're already in `ErrorAlreadyHandled` state and hit the `ErrorAlreadyHandled => nothing` lines, there's nothing more to do because we're already panicking. 

Did I miss a case ?

Reply:
Looks good, thank you!